src/ducks/balance/History.jsx

```diff
@@ -17,22 +17,36 @@ import {
 } from './helpers'
 import { withSize } from 'react-sizeme'
 
+const today = new Date()
+const oneYearBefore = subYears(today, 1)
+
 class History extends Component {
   getBalanceHistory(accounts, transactions) {
-    const today = new Date()
     const balanceHistories = getBalanceHistories(
       accounts,
       transactions,
       today,
-      subYears(today, 1)
+      oneYearBefore
     )
     const balanceHistory = sumBalanceHistories(Object.values(balanceHistories))
 
     return balanceHistory
   }
 
+  getTransactionsFiltered() {
+    const { transactions } = this.props
+
+    return {
+      ...transactions,
+      data: transactions.data.filter(t => {
+        return isAfter(new Date(t.date), oneYearBefore)
+      })
```

Comment:
I have the impression that a redux selector could be used here. It would be easier then to introduce memoization via reselect. Here at each render we are refiltering and doing non trivial operations (date calculation).

Reply:
It's a really good optimisation but I failed to set up reselect. I make it on another PR